### PR TITLE
fetchWeather function Completed

### PR DESCRIPTION
**Pull request by RajeevKumarSinha**
I have taken an array of cities and after fetching all possible cities weather console logging the temperature and wind speed of the city and if there is any error in fetching the data then logging the error.

### CODE REVIEW

File: js-problems/13-weather-api/fetchWeatherData.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  "San Diego",
+  "Dallas",
+  "San Jose",
+];
+
+const fetchWeather = async function (cities, myKey = apiKey) {
+  const citiesWeather = []; // array to store fetched data
+  for (const city of cities) {
+    // loop over cities array
+    const cityWeather = await getJson(
```
Comment (rohandhamapurkar) [inline js-problems/13-weather-api/fetchWeatherData.js]: here on line 50 you are awaiting the response for each city and then proceeding forward is there any way to make this faster?

**Review by rohandhamapurkar — CHANGES_REQUESTED** [review]

File: js-problems/13-weather-api/fetchWeatherData.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  "San Jose",
+];
+
+const fetchWeather = async function (cities, myKey = apiKey) {
+  //////////////////////////////////////////////old(__this code make api calls one at a time__) {
+
+  // const citiesWeather = []; // array to store fetched data
+  // for (const city of cities) {
+  //   // loop over cities array
+  //   const cityWeather = await getJson(
```
Comment (RajeevKumarSinha) [inline js-problems/13-weather-api/fetchWeatherData.js:52]: Screenshot (36)

Comment (RajeevKumarSinha) [reply js-problems/13-weather-api/fetchWeatherData.js:52]: This is the  previous code which makes  api calls sequentially or one at a time.

File: js-problems/13-weather-api/fetchWeatherData.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  const urlCities = cities.map((city) =>
+    getJson(
+      `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&appid=${myKey}`
+    )
+  );
+
+  const [...citiesWeather] = await Promise.all(urlCities);
+  console.log(citiesWeather);
+  citiesWeather.forEach((weather, index) => showWeather(weather, index)); // calling showWeather fn to display fetched data.
+
```
Comment (RajeevKumarSinha) [inline js-problems/13-weather-api/fetchWeatherData.js:72]: Screenshot (37)

Comment (RajeevKumarSinha) [reply js-problems/13-weather-api/fetchWeatherData.js:72]: New code, on line no. 63 added  urlCities array to store the getJson function as callback for every city,  then on line 69 packed citiesWeather array using rest operator to store objects as response from the calling of Promise.all for urlCities array, here Promise.all executes all the callbacks present in urlCities in parallel.

**Review by RajeevKumarSinha — COMMENTED** [review]
I have made the changes in the code for making it faster.

File: js-problems/13-weather-api/fetchWeatherData.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  //////////////////////////////////////////////new(__this code make all the apiCalls at a time__) (){
+
+  const urlCities = cities.map((city) =>
+    getJson(
+      `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&appid=${myKey}`
+    )
+  );
+
+  const [...citiesWeather] = await Promise.all(urlCities);
```
Comment (rohandhamapurkar) [inline js-problems/13-weather-api/fetchWeatherData.js]: Explore just writing `citiesWeather` instead of `[...citiesWeather]`

**Pull request merged**